### ISSUE TITLE
Wishlist: Suggest plugins

### ISSUE THREAD

**Issue by probonopd** [linuxdeploy/linuxdeploy#23]
If I as a user run linuxdeploy e.g., on a Qt application, linuxdeploy should suggest to download the Qt plugin.

**Comment by probonopd** [comment]
Even better, have it download for me automatically.

**Comment by TheAssassin** [comment]
This would require some hard-coded patterns in linuxdeploy which are matched to files in the AppDir, I assume, analog to how plugins shall be enabled automatically at some point.

I think it makes sense to set up some global list of plugins which will return the name and the pattern list for registered plugins. Then, we don't need to hard-code anything.

Regarding auto-download, I think that's a bit dangerous, I wouldn't really want to automate this part. But we can discuss this in a separate issue, once the plugin list is ready.

**Comment by probonopd** [comment]
My purpose is to make 2 lines of code in e.g., `.travis.yml` to 1 line.

**Comment by TheAssassin** [comment]
That'd be premature optimization right now IMO. We should really focus on this "directory of plugins" thingy I suggested first, then use the information from there.

**Comment by probonopd** [comment]
I don't want to have to re-send 1,001 pull requests once two lines can become one line. So I'd rather like to have it one line right from the beginning.

**Comment by TheAssassin** [comment]
This tool is plugin-based, and it will always _be_ plugin based. Using two lines is not an issue IMO, but a feature. We could opt not to ship anything but the appimage plugin, as we do now. That way, your PRs will continue to work. (Not to mention that "external" plugins take precedence over the bundled ones, so it won't ever "become one line", downloading the plugin yourself will always be possible, even if it only was to use a more recent version or a development version to test something). I'd prefer this. We could write one of your beloved PR `.travis.yml` templates in which you just need to comment in all the plugins that shall be used.

**Comment by probonopd** [comment]
You know, my goal is to bring down the whole travis,yml to one `script:` line (everything fully automagic), especially after reading this.

![](https://user-images.githubusercontent.com/2480569/44678981-4272c900-aa29-11e8-8128-80b110b0717d.png)

Greetings @onli ;-)
